x/concentrated-liquidity/concentrated-pool/codec.go

```diff
@@ -0,0 +1,32 @@
+package concentrated_pool
+
+import (
+	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+// RegisterLegacyAminoCodec registers the necessary x/concentrated-liquidity interfaces and concrete types
+// on the provided LegacyAmino codec. These types are used for Amino JSON serialization.
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&Pool{}, "osmosis/concentratedliquidity/ConcentratedPool", nil)
+}
+
+func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	registry.RegisterInterface(
+		"osmosis.concentratedliquidity.v1beta1.PoolI",
+		(*types.PoolI)(nil),
+		&Pool{},
+	)
+}
+
+var (
+	amino     = codec.NewLegacyAmino()
+	ModuleCdc = codec.NewAminoCodec(amino)
+)
```

Comment:
nit: let's keep var definitions before function definitions

Reply:
I just wanted to keep the same format with all the other codec files + I think this structure has a point since we can see the variables right before init wdyt?